### PR TITLE
Launchpad: Adjust Logstash ref to fix fatal on WoA sites

### LINKED ISSUE #31283: jetpack-mu-wpcom: fatal on WoA sites from Launchpad code
On my test WoA site, I've received the following fatal error:

```
[08-Jun-2023 19:20:25 UTC] PHP Fatal error:  Uncaught Error: Failed opening required '/srv/htdocs/wp-content/lib/log2logstash/log2logstash.php' (include_path='/:.') in /srv/htdocs/wp-content/plugins/jetpack-mu-wpcom-plugin-dev/vendor/automattic/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php:261
```

Looks like that was added [here](https://github.com/Automattic/jetpack/pull/31121/files#diff-3e92c2cff4e2dd4ec2e0ef889cac1c73a78275f711d0cd9b4e419b7753de513eR255) in #31121

For WoA sites, `require_once WP_CONTENT_DIR . '/lib/log2logstash/log2logstash.php';` will not be available.

This change did not make it into the package version currently active in wpcomsh on WoA sites, but it will be a blocker for next week's weekly release.

WoA testing tips for `jetpack-mu-wpcom` can be found on internal FG: PCYsg-Osp-p2

CC: @sixhours @andres-blanco @nelsonec87 Could one of you please have a look at fixing this, thanks!

### PR DESCRIPTION
**Pull request by sixhours**
Fixes #31283

## Proposed changes:

* Wrap references to Logstash in an `IS_WPCOM` check to prevent fatal errors on WoA sites, and return WP_Error if task initialization fails for any site.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Create a WoA developer site if you don't have one already
* Use the Jetpack Beta Tester plugin to apply this patch to your WoA site
* Alternatively, apply this patch to the site via SSH/rsync or SFTP; you may need to check out this patch locally and overwrite the entire contents of the `/launchpad` directory, not just this patched file. You'll find the files in `/wpcomsh/vendor/automattic/jetpack-mu-wpcom/src/features/launchpad/`
* `tail -F /tmp/php-errors` on the WoA site
* Ensure you no longer see fatal errors like the following:

Screen Shot 2023-06-08 at 4 01 21 PM

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php [hunk 1]
```diff
@@ -258,28 +258,14 @@ function wpcom_mark_launchpad_task_complete( $task_id ) {
  * @param array $task_definitions The tasks to initialize.
  */
 function wpcom_launchpad_init_listeners( $task_definitions ) {
-	require_once WP_CONTENT_DIR . '/lib/log2logstash/log2logstash.php';
```
Comment (samiff) [inline projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php:261]: If you still wanted to do this on simple wpcom for now, could you use a file_exists check before the require and function_exists check in the catch?

**Comment by samiff** [comment]
Verified this will correct the issue. If we want to merge as-is, will need to add a changelog and fix the lint error about empty catch block.

File: projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php (hunk 1 above)
Comment (sixhours) [reply projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php:261]: Is the `IS_WPCOM` global defined on Atomic sites? That might be a simpler check... looks like that's what we do in `launchpad.php`, though that's also filtering by blog ID.

Comment (samiff) [reply projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php:261]: Ah yes, `if ( defined( 'IS_WPCOM' ) && IS_WPCOM )` would work for checking simple wpcom sites. That constant shouldn't be on Atomic sites, it's how the status package checks for example:

https://github.com/Automattic/jetpack/blob/b5e05139090411f835804a3a2da0c18be9b9e57a/projects/packages/status/src/class-host.php#L61-L68

**Comment by sixhours** [comment]
I'm headed out for the weekend, @nelsonec87 or @andres-blanco please feel free to commandeer this PR and get it shipped to avoid blocking deployment to WoA. Thank you!!

**Review by samiff — APPROVED** [review]
Looks good, thanks for the quick fix @sixhours 🙇

**Pull request merged**